### DISCUSSION TITLE
Digits and analog Model cannot be selected.

### DISCUSSION THREAD

**Discussion by New-three** [jomjol/AI-on-the-edge-device#1470] [General] [answered]
![al on tge](https://user-images.githubusercontent.com/99345562/205332592-d3aa9112-58f2-4dd4-bbbc-8305c307e775.jpg)
### The Problem

I can't select the model for Digits and Analog in the settings. I also get a message in the browser. 
In the selected field the value '/config/ana-cont_11.3.0_s2.tflite' in the section 'Analog' in the field 'Model' is invalid. PLEASE CHECK BEFORE SAVING!

I tried a fresh install. Also a web installation and two SD cards

### Version

12.0.1

### Logfile

```shell
2022-12-02T15:56:04: Time zone set to CET-1CEST,M3.5.0,M10.5.0/3
2022-12-02T15:56:04: =============================================================================================
2022-12-02T15:56:04: =================================== Main Started ============================================
2022-12-02T15:56:04: =============================================================================================
2022-12-02T15:56:04: Branch: 'HEAD', Tag: 'v12.0.1', Revision: dfe2c46, Date/Time: 2022-09-29 17:13
2022-12-02T16:45:19: Time zone set to CET-1CEST,M3.5.0,M10.5.0/3
2022-12-02T16:45:19: =============================================================================================
2022-12-02T16:45:19: =================================== Main Started ============================================
2022-12-02T16:45:19: =============================================================================================
2022-12-02T16:45:19: Branch: 'HEAD', Tag: 'v12.0.1', Revision: dfe2c46, Date/Time: 2022-09-29 17:13
```

**Comment by jomjol** [comment]
Either your /config folder does not contain a tflite / tfl file or you have a mismatch between html version and firmware.

**Comment by New-three** [comment]
I installed version 12.0.1 via the web installer. Then all files from the source code (zip) of version 12.0.1 copied to the SD card.

![al on](https://user-images.githubusercontent.com/99345562/205339290-ff31cf47-0276-443e-b17e-80aaea98642b.jpg)

**Comment by jomjol** [comment]
Can you switch to version 13.x?

**Comment by caco3** [comment]
Most likely a caching issue of your webbrowser!
Try to force-refresh or use another webbrowser

**Comment by New-three** [comment, marked as answer]
I tried the OTA update 13.0.1 again and now it works. 
Many thanks for the replies and great praise for the great project

**Comment by svenschedlbauer** [comment]
Hello, I have exactly the same problem.
How can I update to 13.0.3? I have already downloaded the update.zip file  --> I was able to make the update.
But since that I see no picture in the overview....

**Reply by caco3** [reply]
does the log say something (menu `System > Log Viewer`)?

Maybe a reboot helps?

**Reply by svenschedlbauer** [reply]
Log says:
 [       7] 2022-12-03T21:22:08 <WRN> [MAIN] Web UI version (e9e1358) does not match firmware version (e9e1358+) !
[       9] 2022-12-03T21:22:10 <INF> [MAIN] Initialization completed successfully!
[       9] 2022-12-03T21:22:10 <WRN> [TFLITE] Restarted due to an Exception/panic! Postponing first round start by 5 minutes to allow for an OTA or to fetch the log!
[       9] 2022-12-03T21:22:10 <WRN> [TFLITE] Setting logfile level to DEBUG until the next reboot!
[      15] 2022-12-03T21:22:15 <DBG> [MAIN SERVER] info_get_handler
[      15] 2022-12-03T21:22:15 <DBG> [MAIN SERVER] info_get_handler
[      15] 2022-12-03T21:22:15 <DBG> [MAIN SERVER] info_get_handler
[      15] 2022-12-03T21:22:15 <DBG> [OTA FILE] log_get_last_part_handler

**Reply by caco3** [reply]
> 2022-12-03T21:22:08 [MAIN] Web UI version (https://github.com/jomjol/AI-on-the-edge-device/commit/e9e13588f67c005ce95d42fad6935877d1840519) does not match firmware version (https://github.com/jomjol/AI-on-the-edge-device/commit/e9e13588f67c005ce95d42fad6935877d1840519+) !

This just got fixed in rolling, but its not really a bug

**Reply by svenschedlbauer** [reply]
But what can be the reason? I csn slso not make a new reference picture.
![Screenshot_20221203_224215_Chrome](https://user-images.githubusercontent.com/110197932/205463319-77ee09f3-4e71-4c63-ae59-22d2ae999555.jpg)
![Screenshot_20221203_224259_Chrome](https://user-images.githubusercontent.com/110197932/205463328-6308315a-11f5-4445-83e8-0f7013d9c73e.jpg)

**Reply by caco3** [reply]
does the log show something after you called those pages?
Does it happen after every reboot?

**Reply by svenschedlbauer** [reply]
Don't know, because at the moment I can not open the ip. He is only loading.... Restart does also not help. Will try it tomorrow again.

**Reply by svenschedlbauer** [reply]
Logfile is showing nothing after I called those pages :-(
Yes, Reboot does not change anything. Ot is happening still.

**Reply by caco3** [reply]
please try reinstalling (also setting up the SD-Card).

And use the latest release!

**Reply by svenschedlbauer** [reply]
It is working in the meanwhile. Thank you!

**Reply by stefanbode** [reply]
same problem here after upgrade from 12.0.1 to 13.0.4. Then again additional update then it works partially. The model is back. After that MQTT and Homeassistant also produces error. I assume the default values are the problem. After selecting something from the drop down this is also gone. Remaining problem: Overview page is often empty. The later on no response at all on the ip for 10 minutes. Then it is back. all quite strange. I start from scratch and even here I had to do an update.zip to get back to a more or less working html page. Looks like there is quite a lot messed up now.

**Reply by svenschedlbauer** [reply]
I have seen now, that I have only updaded the file update.zip, which is not correct.
After I updated first the "firmware.bin" file out of the update.zip folder and second I updated the "update.zip" file it seem to work perfect!
May you made the same mistake. Pls check.

**Reply by stefanbode** [reply]
Same procedure. First update zip with a mess. Then firmware update. Still not good but better. Then update.zip again.  Now even better but as mentioned far from perfekt because of missing connections. Need to investigate.

**Reply by svenschedlbauer** [reply]
I am really happy with the result. It is now working great together with my home assistant.

**Reply by caco3** [reply]
> Same procedure. First update zip with a mess. Then firmware update. Still not good but better. Then update.zip again.

The update from 12.0.1 is buggy, this is why we clearly note that you should check that the firmware **and** the Web UI have the correct version. If they mismatch, even a message is shown!

But I agree that during adding all those new functionality, some bits an d pieces got broken. We work hard to fix it all again, please be patient!
